### PR TITLE
Refs #10176,#9753 - container name is optional, BZ1204927

### PR DESCRIPTION
**Pull request by adamruzicka**
If a name is not supplied, it uses the one generated by docker without the leading /

### CODE REVIEW

File: app/models/service/containers.rb [hunk 1]
```diff
@@ -45,6 +45,9 @@ def create_container_object(wizard_state)
 
       fail ActiveRecord::Rollback unless pull_image(container) && start_container(container)
 
+      container.name = container.in_fog.name[1..-1] if container.name.nil? ||
```
Comment (eLobato) [inline app/models/service/containers.rb]: You can simply do `if container.name.present?`

**Comment by eLobato** [comment]
The commit name should rather be "Fixes #9753", I closed the other bug as a duplicate. For some reason the name is not saved as the container name. This leads to the following issue:

1. I create a container without name - Docker assigns it the name "agitated_morse", but foreman-docker didn't assign that name to it.
2. I try to create another container without name. I can't because now there's a validation for unique names of containers in a compute resource.

Code looks good except for that tiny inline comment :smile: 

Please look into what could be happening so that the name is not saved, and it'd be in! Thanks again @adamruzicka !

**Comment by adamruzicka** [comment]
@eLobato Sorry for the delay. I tried looking into it but I couldn't reproduce it. Works for me when creating in the web UI as well as using hammer. Any ideas how to reproduce it?

**Comment by eLobato** [comment]
I guess I was doing something differently, it does work. Merging in 3..2..1.. :bomb:

**Comment by eLobato** [comment]
Merged as f9547788b5f504a4328b5c65e28ab111b3cf46f4, thank you @adamruzicka

**Pull request closed without merging**